app/models/spree/admin/tabs/availability_builder_methods.rb

```diff
@@ -0,0 +1,27 @@
+module Spree
+  module Admin
+    module Tabs
+      module AvailabilityBuilderMethods
+        def with_availability_check(availability_check)
+          @availability_check = availability_check
+          self
+        end
+
+        def with_update_availability_check
+          @availability_check = ->(ability, resource) { ability.can?(:update, resource) }
+          self
+        end
+
+        def with_index_availability_check(klass)
+          @availability_check = ->(ability, _resource) { ability.can?(:index, klass) }
+          self
+        end
+
+        def with_admin_availability_check(klass)
```

Comment:
I think we have the same methods in the mixin for the main menu. Maybe it would make sense to move the AvailabilityBuilderMethods to the Spree::Admin module (or Spree::Admin::Permissions or something like that), and then include the same mixin both in main menu and tabs?

Reply:
I'll resolve this in #276 to save myself rebasing 😉

Reply:
Changed my mind, I'll do it here.